File: ProjC1/src/projc1/Game/RenderEngine.java
package projc1.Game;

class RenderEngine {
    public void drawRectangle(java.awt.Graphics2D g2d, int x, int y, int width, int height, java.awt.Color color) {
        g2d.setColor(color);
        g2d.fillRect(x, y, width, height);
    }
    
    public void drawOval(java.awt.Graphics2D g2d, int x, int y, int width, int height, java.awt.Color color) {
        g2d.setColor(color);
        g2d.fillOval(x, y, width, height);
    }
    
    public void drawBorder(java.awt.Graphics2D g2d, int x, int y, int width, int height, java.awt.Color color) {
        g2d.setColor(color);
        g2d.drawRect(x, y, width, height);
    }
    
    public void drawText(java.awt.Graphics2D g2d, String text, int x, int y, java.awt.Font font, java.awt.Color color) {
        g2d.setColor(color);
        g2d.setFont(font);
        g2d.drawString(text, x, y);
    }
}

File: ProcC1_pyver/game/components/render_engine.py
class RenderEngine:
    def draw_oval(self, canvas, x, y, width, height, color):
        canvas.create_oval(x, y, x + width, y + height, fill=color, outline="")
    
    def draw_border(self, canvas, x, y, width, height, color):
        canvas.create_rectangle(x, y, x + width, y + height, outline=color)
    
    def draw_text(self, canvas, text, x, y, font, color):
        canvas.create_text(x, y, text=text, font=font, fill=color)